src/drivers/intel/dmic.c

```diff
@@ -1104,10 +1104,13 @@ static int dmic_get_hw_params(struct dai *dai,
 	return 0;
 }
 
-static int dmic_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
+static int dmic_set_config(struct dai *dai, struct ipc_config_dai *common_config,
+			   void *spec_config)
 {
 	struct sof_ipc_dai_dmic_params **uncached_dmic_prm = cache_to_uncache(&dmic_prm[0]);
 	struct dmic_pdata *dmic = dai_get_drvdata(dai);
+	struct sof_ipc_dai_config *config =
+				(struct sof_ipc_dai_config *)spec_config;
```

Comment:
@lgirdwood maybe we dont need the typecasting here in the other dai_type.c files no?

Reply:
Yeah, I originally had another type here, no need for cast.